lib/errordedup/errordedup.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package errordedup
```

Comment:
`errordedup` seems slightly misnamed, as we are not dealing specifically with errors but with deduping/rate limiting logs.

Reply:
Renamed to `loglimit` in c56f31e04b2349e1bf6666e86603e64562a8b8a6.